### PR TITLE
[WFCORE-4486] Support for multiple security realms - Failover

### PR DESCRIPTION
**Pull request by nekdozjam**
https://issues.jboss.org/browse/WFCORE-4486

### CODE REVIEW

File: elytron/WFCORE-4486-multiple-security-realms-failover.adoc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+`failover-realm` will emit SecurityRealmUnavailableEvent to the corresponding SecurityDomain whenever it fails over. This can be turned off by adding `emit-events` attribute and setting it to `false`.
+
+=== Nice-to-Have Requirements
+
+
+
+== Test Plan
+
+Tests will be added to Wildfly Elytron
```
Comment (OndrejKotek) [inline elytron/WFCORE-4486-multiple-security-realms-failover.adoc:71]: Can you provide specific cases please?

File: elytron/WFCORE-4486-multiple-security-realms-failover.adoc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+=== Affected Projects or Components
+
+* Wildfly Core
+* Elytron
+
+== Requirements
+
+=== Hard Requirements
+
+Add FailoverRealm to Elytron, which will be a SecurityRealm wrapper containing delegate and failover realm. If the delegate throws a RealmUnavailableException during RealmIdentity lookup, it will be caught and failover realm will be used instead. Configuration in WFCore will look like following:
```
Comment (OndrejKotek) [inline elytron/WFCORE-4486-multiple-security-realms-failover.adoc:53]: Do we really want to allow just one failover realm? Why not a list of failover realms?

File: elytron/WFCORE-4486-multiple-security-realms-failover.adoc [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
```
Comment (OndrejKotek) [inline elytron/WFCORE-4486-multiple-security-realms-failover.adoc:45]: What about WildFly (Documentation and Tests Only)?

File: elytron/WFCORE-4486-multiple-security-realms-failover.adoc [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+=== Nice-to-Have Requirements
+
+
+
+== Test Plan
+
+Tests will be added to Wildfly Elytron
+
+
```
Comment (OndrejKotek) [inline elytron/WFCORE-4486-multiple-security-realms-failover.adoc:73]: What about tests in WildFly Core for the new management model (the parsing and persisting of configuration, transformer testing).

File: elytron/WFCORE-4486-multiple-security-realms-failover.adoc (hunk 2 above)
Comment (nekdozjam) [reply elytron/WFCORE-4486-multiple-security-realms-failover.adoc:53]: This is meant mostly as a backup solution (eg. db is down, so lets use something file based as a backup for admins to get in), so I dont see the scenario of multiple failover realms common. But having multiple failover realms is still possible by simply stacking the realms.

File: elytron/WFCORE-4486-multiple-security-realms-failover.adoc (hunk 4 above)
Comment (nekdozjam) [reply elytron/WFCORE-4486-multiple-security-realms-failover.adoc:73]: added

**Pull request merged**